uni/udb/breakpoint.icn

```diff
@@ -508,121 +593,170 @@ method cmdClear(cmd)
+      return
+   }
    else if id := integer(cmd[3]) then {
       every x := key(breakPoints) do {
-	 L := breakPoints[x]
-	 every i := 1 to *L do {
-	    if L[i].id = id then
-	       if L[i].state ~= DELETED then {
-	          L[i].state := DELETED
-	          msg := "\n   Breakpoint #"||id||" is deleted."
-	          DState.Write(msg)
-	          return
-	          }
-	       else {
-	          msg := "\n   Breakpoint #"||id||" is already deleted."
-	          DState.Write(msg)
-	          fail
-	         }
-	    }
-	 }
-      msg := "\n   Breakpoint #"||id||" is not available."
-      DState.Write(msg)
-      fail
-      }
+         L := breakPoints[x]
+         every i := 1 to *L do {
+            if L[i].id = id then
+               if L[i].state ~= DELETED then {
+                  L[i].state := DELETED
+                  msg := "\n   Breakpoint #"||id||" is deleted."
+                  resultTable["success"] := "__true__"
+                  resultTable["message"] := "Breakpoint #"||id||" is deleted"
+                  resultTable["consoleMsg"] := msg
+                  DState.Write(resultTable)
+                  return
+               }
+               else {
+                  msg := "\n   Breakpoint #"||id||" is already deleted."
+                  resultTable["success"] := "__false__"
+                  resultTable["message"] := "Breakpoint #"||id||" is already deleted"
+                  resultTable["consoleMsg"] := msg
+                  DState.Write(resultTable)
+                  fail
+               }
+	      }
+	   }
+   msg := "\n   Breakpoint #"||id||" is not available."
+   resultTable["success"] := "__false__"
+   resultTable["message"] := "Breakpoint #"||id||" is not available"
+   resultTable["consoleMsg"] := msg
+   DState.Write(resultTable)
+   fail
+   }
    else {
       msg:="\n   Undefined Command: \""||DState.cmdHistory[1]||"\""||
 	   "\n   Try \"help\" for assistance."
-      DState.Write(msg)
+      resultTable["success"] := "__false__"
+      resultTable["message"] := "Undefined Command: "||DState.cmdHistory[1]
+      resultTable["consoleMsg"] := msg
+      DState.Write(resultTable)
       fail
-      }
+   }
 end
 
 #
 # It deletes pre-set breakpoints
 #
 method cmdEnable(cmd)
    local id, x, L, i, j:=0
+   local resultTable := table()
 
    # Enables all the preset breakpoints
    # enable break
    if not cmd[3] then {
       every x := key(breakPoints) do {
-	 L := breakPoints[x]
-	 every i := 1 to *L do
-	    if L[i].state = DISABLED then {
-	       L[i].state := ENABLED
-	       j +:= 1
-	       }
-	 }
-      if j > 0 then
-	 msg := "\n   All breakpoints are enabled; ("||j||" total)."
-      else
-	 msg := "\n   No disabled breakpoints are available to be enabled."
-      DState.Write(msg)
-      return
+         L := breakPoints[x]
+         every i := 1 to *L do {
+            if L[i].state = DISABLED then {
+               L[i].state := ENABLED
+               j +:= 1
+            }
```

Comment:
Since you don't care about the index, and only operating on breakpoints, I suggest writing such code as : 
```
         every b := !L do {
            if b.state = DISABLED then {
               b.state := ENABLED
               j +:= 1
            }
```

Reply:
Implemented